linearregression/Lib.idr

```diff
@@ -0,0 +1,51 @@
+module Lib
+import Types
+
+
+public export
+calculateDeltas : (Double,Double) -> (Double, Double) -> Double
+calculateDeltas (t0, t1) (x, y) = t0 + t1 * x-y
+
+
+
+avg : List Double -> Double
+avg [] = 0.0
+avg xs = do
+  let x = length xs
+  let y = natToInteger x
+  let z = fromInteger y
+  (sum xs) / z
```

Comment:
BTW, I've just discovered that most of the time you can just use `cast` (defined in Prelude.Cast) and idris will do the right thing.

Reply:
@ngeiswei  Yeah I looked at that too while trying to do the casting. not sure though how i can use it can you give me an example invoking it.